src/middlewared/middlewared/plugins/pool.py

```diff
@@ -525,6 +536,20 @@ def pool_extend(self, pool):
         Str('name', required=True),
         Bool('encryption', default=False),
         Str('deduplication', enum=[None, 'ON', 'VERIFY', 'OFF'], default=None, null=True),
+        Dict(
+            'encryption_options',
+            Bool('generate_key', default=False),
+            Bool('key_file', default=False),
+            Int('pbkdf2iters', default=350000, validators=[Range(min=100000)]),
+            Str(
+                'algorithm', default='AES-256-CCM', enum=[
+                    'AES-128-CCM', 'AES-192-CCM', 'AES-256-CCM', 'AES-128-GCM', 'AES-192-GCM', 'AES-256-GCM'
+                ]
+            ),
+            Str('key_format', default='HEX', enum=list(ZFSKeyFormat.__members__)),
```

Comment:
It makes me wonder if we arent just complicating yourselves by giving the user a choice between RAW and HEX? isnt HEX just the RAW but 16bit encoded?

Reply:
To add, uploading a key file sounds a very edge case that we shouldnt probably even expose in the UI. What makes sense is auto generating the key, as we currently do.